Jenkinsfile

```diff
@@ -23,17 +43,80 @@ node('us-east-1 && ubuntu && docker && !gpu') {
                     ./mach package
                 '''
             }
-            stage('Upload APK') {
-                archiveArtifacts allowEmptyArchive: true, artifacts: 'mozilla-release/objdir-frontend-android/dist/*.apk'
+        }
+        withEnv(["INSTANCE_ID=${instance_id}"]){
+            timeout(5){
+                stage('Genymotion Status'){
+                    def status = sh(returnStdout: true, script: """
+                            aws ec2 describe-instance-status --region=us-east-1 --instance-id $INSTANCE_ID --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' --output text
+                        """).trim()
+                    while (status != 'passed') {
+                        println "Waiting for the instance to fully Boot up...."
+                        sleep(15)
+                        status = sh(returnStdout: true, script: """
+                            aws ec2 describe-instance-status --region=us-east-1 --instance-id $INSTANCE_ID --query 'InstanceStatuses[].InstanceStatus[].Details[].Status' --output text
+                        """).trim()
+                        println "Instance Status: ${status}"
+                    }
+                }
             }
         }
-        finally {
-            stage('Clean Up') {
-                sh '''#!/bin/bash -l
-                    rm -f mozilla-release/mozconfig
-                    rm -rf mozilla-release/objdir-frontend-android
-                '''
+        docker.image("141047255820.dkr.ecr.us-east-1.amazonaws.com/${baseImageName}").inside {
+            withEnv(["IP=${ip}", "APP=${apk}", "platformName=android", "deviceName=127.0.0.1:5556"]){
+                withCredentials([file(credentialsId: 'e87da5bf-eb00-4d85-8ef9-93680abaa266', variable: 'FILE' )]){
```

Comment:
what is this FILE exactly?

Reply:
Credentials To connect to the Genymotion Instance.